soc: silabs: siwx91x: add missing kconfig for ns16550 driver variant

Martinhoff-maker opened the pull request:
The ns16550 UART driver has multiple variant Kconfig options.

To successfully run the uart_basic_api test, the correct variant must be selected. 

This commit adds the missing Kconfig options to the siwx917_rb4338a and siwx917_rb4342a boards.

I don't find the fix really clean but the proper way would be to have a different compatible.

Martinhoff-maker committed:
boards: silabs: add missing tag for uart on siwx917 board

Add missing uart tag for siwx91x boards.

Signed-off-by: Martin Hoff <martin.hoff@silabs.com>

jhedberg commented on boards/silabs/radio_boards/siwx917_rb4342a/siwx917_rb4342a_defconfig:
```diff
@@ -11,3 +11,6 @@ CONFIG_MEMC=y
 CONFIG_CLOCK_CONTROL_INIT_PRIORITY=28
 CONFIG_MEMC_INIT_PRIORITY=29
+
+# Variant for the ns16550 driver
+CONFIG_UART_NS16550_DW8250_DW_APB=y
```
Same here. IMO we should modify this with `Kconfig.defconfig` instead, i.e. something like:
```
configdefault UART_NS16550_DW8250_DW_APB
	default y
```

The above should retain the dependency on `UART_NS16550`, making it still possible to completely disable UART support for a build.

Martinhoff-maker replied:
I have hesitate between those two options but only with config and not configdefault (always forgot it exists). With configdefault and inheritance of the dependancy, that's clearly the right things to do. Changed.

Martinhoff-maker committed:
soc: silabs: siwx91x: add missing kconfig for ns16550 driver variant

The ns16550 UART driver has multiple variant Kconfig options. To
successfully run the uart_basic_api test, the correct variant must be
selected. This commit adds the missing Kconfig option.

Signed-off-by: Martin Hoff <martin.hoff@silabs.com>

Martinhoff-maker commented:
v2: move to Kconfig.defconfig
